components/brave_wallet/browser/internal/hd_key.cc

```diff
@@ -88,25 +89,7 @@ bool UTCDecryptPrivateKey(const std::string& derived_key,
 }  // namespace
 
 HDKey::HDKey()
-    : depth_(0),
-      fingerprint_(0),
-      parent_fingerprint_(0),
-      index_(0),
-      identifier_(20),
-      private_key_(new std::vector<uint8_t>(0),
-                   SecureZeroVectorDeleter<uint8_t>()),
-      public_key_(33),
-      chain_code_(32),
-      secp256k1_ctx_(secp256k1_context_create(SECP256K1_CONTEXT_SIGN |
-                                              SECP256K1_CONTEXT_VERIFY)) {}
-HDKey::HDKey(uint8_t depth, uint32_t parent_fingerprint, uint32_t index)
-    : depth_(depth),
-      fingerprint_(0),
-      parent_fingerprint_(parent_fingerprint),
-      index_(index),
-      identifier_(20),
-      private_key_(new std::vector<uint8_t>(0),
-                   SecureZeroVectorDeleter<uint8_t>()),
+    : identifier_(20),
```

Comment:
Parametrized ctor still wasn't enough to create a valid hdkey(needed further setup), so keep only simple HDKey() ctor. depth_, fingerprint_, etc int are header initialized.